MTM-51837-missing-field-limitation-for-group-description (#978)
SoftwareAG/c8y-docs: content/concepts/domain-model-bundle/inventory.md

For more information on fragments and how managed objects are structured, see the [Device management library](reference/device-management-library/) in the *Reference guide*.

{{< c8y-admon-info >}}
While designing the data model for the inventory Managed Object it's worth to keep in mind following:

> 1. There is no size or length constrains for single fragment, but there is a limitation for overal json document size, which can not exceed 16MB for single Managed Object entry within inventory collection.

Reviewer: The limit is 16 MiB, but I would say we should try to keep managed objects not bigger than 1 MiB

Author: fixed

> 2. Each consecutive fragment added to Managed Object at root level will impose certain delay on querying such an item. If performance of query matters, it's better to nest custom fragments with information within chosen single fragment effectively limiting root fragments number. For example:

Reviewer 2: Do we really have such problem as described in 2? Fragments are indexed, so my guess would be it is just about size of payload (and size of indexes in DB), but if you embedd something, you will not be able to search by it efficiently.

I know the different problem --> **size** of the **database array**, which means number of elements inside json list/array `[    ]`. 

This for sure is visible if we try to add more than 10K elements to the same group. DB consultant suggested no more than 1K elems in array for optimal performance, but I think up to 10K may be acceptable. Our tests shown that trying to add 40-50K elements to the single group may end up in each request processing for 1-2 minutes, even though we did some optimizations in this are!

Reviewer: Not bigger than 1K when there is no index on array :) With index the size should be even smaller

For more information on fragments and how managed objects are structured, see the [Device management library](reference/device-management-library/) in the *Reference guide*.

{{< c8y-admon-info >}}
While designing the data model for the inventory managed object consider the following:

> 1. There is no size or length constraint for a single fragment, but there is a limitation for the overall JSON document size, which may not exceed 16MiB for single managed object entry within the inventory collection. We recommend you to keep it below 1 MiB.

Reviewer 3: ```suggestion
1. There is no size or length constraint for a single fragment, but there is a limitation for the overall JSON document size, which may not exceed 16MiB for a single managed object entry within the inventory collection. We recommend you to keep it below 1 MiB.
```

> 1. There is no size or length constrains for single fragment, but there is a limitation for overal json document size, which can not exceed 16MiB for single Managed Object entry within inventory collection. Our recomendation is to try keeping it below 1 MiB.

Reviewer 3: ```suggestion
1. There is no size or length constraint for a single fragment, but there is a limitation for the overall JSON document size, which may not exceed 16MiB for single managed object entry within the inventory collection. We recommend you to keep it below 1 MiB.
```

There is no size or length constraint for a single fragment, but there is a limitation for the overall JSON document size, which may not exceed 16MB for a single managed object entry within the inventory collection.

> 2. While including arrays of elements within fragments, be advised to keep length of such collections below 1k elements.

Reviewer 3: ```suggestion
2. When you include arrays of elements within fragments, keep the length of such collections below 1k elements.
```

> 3. Each consecutive fragment added to Managed Object at root level will impose certain delay on querying such an item. If performance of query matters, it's better to nest custom fragments with information within chosen single fragment effectively limiting root fragments number. For example:

Reviewer 3: ```suggestion
3. Each consecutive fragment added to the managed object at root level will impose a certain delay on querying such an item. If the performance of a query matters, it's recommended to nest custom fragments with information within a chosen single fragment effectively limiting the root fragments number. For example:
```

> 3. Each consecutive fragment added to the managed object at root level will impose a certain delay on querying such an item. If the performance of a query matters, it is recommended to nest custom fragments with information within a chosen single fragment effectively limiting the root fragments number. For example:

Reviewer 3: ```suggestion
3. Each consecutive fragment added to the managed object at root level imposes a certain delay on querying such an item. If the performance of a query matters, it is recommended to nest custom fragments with information within a chosen single fragment effectively limiting the root fragments number. For example:
```
